Lib/test/test_dict.py

```diff
@@ -1222,6 +1278,12 @@ def iter_and_mutate():
 
         self.assertRaises(RuntimeError, iter_and_mutate)
 
+    def test_reversed(self):
+        d = {i: i for i in range(20)}
```

Comment:
Because integers have a special `hash()` behavior, perhaps a better test would be to use a few letters: "abcdefg". This shouldn't matter for the current dict implementation, but might make the test more robust against future implementation changes.

```
    def test_reversed(self):
        keys = 'abcdefg'
        d = dict.from_keys(keys)
        r = reversed(d)
        self.assertEqual(list(r), keys[::-1])
        self.assertRaises(StopIteration, next, r)
```